tests/components/switch/test_deconz.py

```diff
@@ -0,0 +1,90 @@
+"""deCONZ switch platform tests."""
+from unittest.mock import Mock, patch
+
+from homeassistant import config_entries
+from homeassistant.components import deconz
+from homeassistant.components.deconz.const import SWITCH_TYPES
+from homeassistant.helpers.dispatcher import async_dispatcher_send
+
+from tests.common import mock_coro
+
+SUPPORTED_SWITCHES = {
+    "1": {
+        "id": "Switch 1 id",
+        "name": "Switch 1 name",
+        "type": "On/Off plug-in unit",
+        "state": {}
+    },
+    "2": {
+        "id": "Switch 2 id",
+        "name": "Switch 2 name",
+        "type": "Smart plug",
+        "state": {}
+    }
+}
+
+UNSUPPORTED_SWITCH = {
+    "1": {
+        "id": "Switch id",
+        "name": "Unsupported switch",
+        "type": "Not a smart plug",
+        "state": {}
+    }
+}
+
+
+async def setup_bridge(hass, data):
+    """Load the deCONZ switch platform."""
+    from pydeconz import DeconzSession
+    loop = Mock()
+    session = Mock()
+    entry = Mock()
+    entry.data = {'host': '1.2.3.4', 'port': 80, 'api_key': '1234567890ABCDEF'}
+    bridge = DeconzSession(loop, session, **entry.data)
+    with patch('pydeconz.DeconzSession.async_get_state',
+               return_value=mock_coro(data)):
+        await bridge.async_load_parameters()
+    hass.data[deconz.DOMAIN] = bridge
+    hass.data[deconz.DATA_DECONZ_UNSUB] = []
+    hass.data[deconz.DATA_DECONZ_ID] = {}
+    config_entry = config_entries.ConfigEntry(
+        1, deconz.DOMAIN, 'Mock Title', {'host': 'mock-host'}, 'test')
+    await hass.config_entries.async_forward_entry_setup(config_entry, 'switch')
+    # To flush out the service call to update the group
+    await hass.async_block_till_done()
+
+
+async def test_no_switches(hass):
+    """Test that no switch entities are created."""
+    data = {}
+    await setup_bridge(hass, data)
+    assert len(hass.data[deconz.DATA_DECONZ_ID]) == 0
+    assert len(hass.states.async_all()) == 0
+
+
+async def test_switch(hass):
+    """Test that all supported switch entities and switch group are created."""
+    await setup_bridge(hass, {"lights": SUPPORTED_SWITCHES})
+    assert "switch.switch_1_name" in hass.data[deconz.DATA_DECONZ_ID]
+    assert "switch.switch_2_name" in hass.data[deconz.DATA_DECONZ_ID]
+    assert len(SUPPORTED_SWITCHES) == len(SWITCH_TYPES)
```

Comment:
This isn't really testing the production code, just that we cover all cases in the tests.

Reply:
Later on there is gonna be more types of switches than power plugs so I want to make sure the lists are correct